### PR TITLE
Provide reverse-requires search in zypper

### LINKED ISSUE #214: Provide reverse-requires search in zypper
Just as apt has a command called "apt-rdepends" that you can use to find all packages depending on a given package, we should have a option in zypper search (maybe reverse-requires) that allows the user to find all packages that require a given package. 

For example:
zypper se --reverse-requires libqt5core5  

Also we should have a switch wether to expand packages (i.e. 'FOO' as 'literally FOO' or 'everything FOO provides' in case it's a package)

### PR DESCRIPTION
**Pull request by bzeller**
This patch adds a new set of switches to zypper to support searching
reverse dependencies for a package or a set of packages (fixes #214)

### CODE REVIEW

File: doc/zypper.8.txt [hunk 1]
```diff
@@ -777,6 +777,27 @@ Query Commands
...
 		Search for packages which supplement the search strings.
 
+	*--provides-pkg*::
+		Search for all packages that provide any of the provides of the package(s) matched by the input parameters.
+
+	*--requires-pkg*::
+		Search for all packages that require any of the provides of the package(s) matched by the input parameters.
+
+	*--recommends-pkg*::
+		Search for all packages that recommends any of the provides of the package(s) matched by the input parameters.
```
Comment (mlandres) [inline doc/zypper.8.txt]: `Search for all packages that recommend`, without `s`?
Same in the descriptions below.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -11,6 +11,7 @@ MESSAGE ( "** Locale dir is ${INSTALL_PREFIX}/share/locale" )
 
 INCLUDE_DIRECTORIES(${ZYPPER_SOURCE_DIR}/src)
 
+
```
Comment (mlandres) [inline src/CMakeLists.txt]: Superfluous insert?

File: src/commands/search/search.cc [hunk 3]
```diff
@@ -156,6 +169,34 @@ zypp::ZyppFlags::CommandGroup SearchCmd::cmdOptions() const
...
+          // translators: --provides-pkg
+          _("Search for all packages that provide any of the provides of the package(s) matched by the input parameters.")
+        },
+        { "requires-pkg", '\0', ZyppFlags::NoArgument, ZyppFlags::setSolvAttrOptional( that._requestedReverseSearch, sat::SolvAttr::requires ),
+          // translators: --requires-pkg
+          _("Search for all packages that require any of the provides of the package(s) matched by the input parameters.")
+        },
+        { "recommends-pkg", '\0', ZyppFlags::NoArgument, ZyppFlags::setSolvAttrOptional( that._requestedReverseSearch, sat::SolvAttr::recommends ),
+          // translators: --recommends-pkg
+          _("Search for all packages that recommends any of the provides of the package(s) matched by the input parameters.")
```
Comment (mlandres) [inline src/commands/search/search.cc]: Same as before: no `s` IMO sounds better.
Same below.

Comment (bzeller) [reply src/commands/search/search.cc]: Done

File: src/CMakeLists.txt (hunk 2 above)
Comment (bzeller) [reply src/CMakeLists.txt]: Done

File: doc/zypper.8.txt (hunk 1 above)
Comment (bzeller) [reply doc/zypper.8.txt]: Done

**Pull request merged**